database/rrd.h

```diff
@@ -1093,16 +1112,15 @@ extern RRDHOST *localhost;
 #define rrdhost_sender_replicating_charts_zero(host) (__atomic_store_n(&((host)->rrdpush_sender_replicating_charts), 0, __ATOMIC_RELAXED))
 
 extern DICTIONARY *rrdhost_root_index;
-long rrdhost_hosts_available(void);
+size_t rrdhost_hosts_available(void);
 
 // ----------------------------------------------------------------------------
-// these loop macros make sure the linked list is accessed with the right lock
 
 #define rrdhost_foreach_read(var) \
-    for((var) = localhost, rrd_check_rdlock(); var ; (var) = (var)->next)
+    for((var) = localhost; var ; (var) = (var)->next)
 
 #define rrdhost_foreach_write(var) \
-    for((var) = localhost, rrd_check_wrlock(); var ; (var) = (var)->next)
+    for((var) = localhost; var ; (var) = (var)->next)
 
```

Comment:
Right now both `rrdhost_foreach_read` and `rrdhost_foreach_write` has the same code, probably we can unify both. 

I will take a look later in the code to verify where they have been called.